src/yoroi-wallets/cardano/ShelleyWallet.ts

```diff
@@ -973,6 +987,32 @@ export class ShelleyWallet implements WalletInterface {
     return api.fetchCurrentPrice(symbol, this.getBackendConfig())
   }
 
+  async fetchNfts(): Promise<YoroiNFT[]> {
+    const utxos = this.utxos
+    const assets = utxos.flatMap((utxo) => utxo.assets ?? [])
+
+    const nftAssets = assets
+      .filter((asset) => asset.policyId && asset.name)
+      .map((asset) => {
+        const {policyId, name} = asset
+        return {nameHex: name, policy: policyId}
+      })
+
+    if (nftAssets.length === 0) {
+      return []
+    }
+
+    return api.getNFTs({assets: nftAssets}, this.getBackendConfig())
+  }
+
+  async fetchNftModerationStatus(fingerprint: string): Promise<YoroiNFTModerationStatus> {
```

Comment:
I'm not really sure if this is the best place to have this function. What are your thoughts?

Reply:
:arrow_up: